sql/core/src/test/scala/org/apache/spark/sql/sources/v2/TableCapabilityCheckSuite.scala

```diff
@@ -91,11 +144,10 @@ class V2WriteSupportCheckSuite extends AnalysisTest {
         EqualTo(AttributeReference("x", LongType)(), Literal(5)))
 
       val exc = intercept[AnalysisException]{
-        V2WriteSupportCheck.apply(plan)
+        TableCapabilityCheck.apply(plan)
       }
 
-      assert(exc.getMessage.contains(
-        "does not support overwrite expression (`x` = 5) in batch mode"))
+      assert(exc.getMessage.contains("does not support overwrite by filter in batch mode"))
```

Comment:
I updated the error message a little bit. The previous message is a little misleading that only the specific expression is not supported.